src/index.js

```diff
@@ -281,26 +301,66 @@ export default class ToggleBlock {
    */
   renderItems() {
     const icon = this.wrapper.firstChild;
-    const editorBlocks = this.api.blocks.getBlocksCount();
+    let toggleRoot;
+
+    if (this.readOnly) {
+      const blocksInEditor = this.api.blocks.getBlocksCount();
+      const currentBlock = this.api.blocks.getCurrentBlockIndex();
+      const initial = blocksInEditor - currentBlock;
+      const redactor = document.getElementsByClassName('codex-editor__redactor')[0];
+      const { children } = redactor;
+      const { length } = children;
+
+      for (let i = initial; i < length; i += 1) {
+        const blockCover = children[i].firstChild;
+        const blockContainer = blockCover.firstChild;
+        const { id } = blockContainer;
+
+        if (id === this.wrapper.id) {
+          toggleRoot = i;
+          break;
+        }
+      }
+    } else {
+      const toggle = this.wrapper.children[1];
+      let currentBlock = {};
 
-    let originalIndex = this.api.blocks.getCurrentBlockIndex();
-    let index = (editorBlocks > 1) ? originalIndex + 1 : originalIndex;
+      while (currentBlock[1] !== toggle) {
+        toggleRoot = this.api.blocks.getCurrentBlockIndex();
+        const block = this.api.blocks.getBlockByIndex(toggleRoot);
+        const { holder } = block;
+        const blockCover = holder.firstChild;
+        const blockContent = blockCover.firstChild;
+        currentBlock = blockContent.children;
+
+        this.api.caret.setToNextBlock('end', 0);
+      }
+    }
+    let index = toggleRoot;
+    index += this.readOnly ? 1 : 0;
 
     icon.addEventListener('click', () => {
       this._resolveToggleAction();
       setTimeout(() => {
-        this._hideAndShowBlocks();
+        const toggleIndex = this.readOnly ? (toggleRoot - 1) : null;
+        this._hideAndShowBlocks(toggleIndex);
       }, 100);
     });
 
     this.data.items.forEach((block) => {
       const { type, data } = block;
-      this.api.blocks.insert(type, data, {}, index += 1, true);
-      this.setAttributesToNewBlock();
+
+      if (this.readOnly) {
+        this.api.blocks.insert(type, data, {}, index, true);
+      } else {
+        this.api.blocks.insert(type, data, {}, index + 1, true);
+      }
```

Comment:
@juliansg022 This code segment may be improved. What do you think?

Reply:
@juanhurtado10 You're right. I'll improve it.